### ISSUE TITLE
`fromBlock: 0` defaults to `latest`, `fromBlock: 1` works correctly in `eth.getPastLogs`

### ISSUE THREAD

**Issue by pszabop** [ethereum/web3.js#1100]
`fromBlock: 0` defaults to `latest`, `fromBlock: 1` works correctly in `eth.getPastLogs`

I suspect somewhere in either web3.js or ethereumjs-testrpc there's a boolean check for `fromBlock` which evaluates to `false` if `fromBlock` is set to zero...

Repro:

1.  create some logs
1. find logs starting at block 0.  Only the latest transaction's logs show up.

```
return eth.getPastLogs({
    fromBlock: 0,
    toBlock: 'latest',
    address: this.contract.address,
    topics: topics,
  }).then(r => {
  // only the latest transaction's logs are in the result
```

1.  find logs starting at block 1.  All prior logs get returned as expected.

```
return eth.getPastLogs({
    fromBlock: 1,
    toBlock: 'latest',
    address: this.contract.address,
    topics: topics,
  }).then(r => {
  // all logs matching topics returned as expected.
```

**Comment by pszabop** [comment]
I note the workaround works, but is very annoying with truffle because you have to issue a bogus transaction to increment the beginning block count to 1 (or else tests fail)

**Comment by varsnotwars** [comment]
What happens when you pass in 0 as a string: "0". That would get around the coercion of 0, if that is what is happening.

**Comment by pszabop** [comment]
@varsnotwars passing in `'0'` doesn't find anything at all...

btw I'm on `    "web3-eth": "^1.0.0-beta.21"`  should I try upgrading?

**Comment by pszabop** [comment]
I believe I have likely tracked down at least one location that has this problem:

https://github.com/MetaMask/provider-engine/blob/c878e5ac19e061dc3de7f317f5ed4090b3fbd49c/subproviders/filters.js#L328

I shall file this bug there... but figuring out the dependency tree is still a work in progress...

**Comment by pszabop** [comment]
I'm not an expert in ethereum javascript dependency trees but it's likely that the `provide-engine` is the source of the bug.  Searching my `npm_module` tree found the above to be the only likely source.

**Comment by brannondorsey** [comment]
This bug through me for a loop for quite a while. Until it's fixed, we should maybe update the new web3.js 1.0 docs to warn users of this unexpected behavior.

**Comment by shivamsinha15** [comment]
+1

**Comment by nivida** [comment]
This issue will be solved in the next build 💪🏻

**Comment by nivida** [comment]
I fixed this in the PR #2000.
